### ISSUE TITLE
Facebook app is crashing when opened by ShareDialog using sdk 16.2.0 or 16.2.1

### ISSUE THREAD

**Issue by TatyanaLeschenok** [facebook/facebook-ios-sdk#2288]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

15.0.0

### Facebook iOS SDK version

16.2.1, 16.2.0

### Dependency Manager

Prebuilt Binaries

### SDK Framework

Share

### Goals

The issue to be resolved wether on the SDK side or Facebook app side.

### Expected results

Facebook iOS app opens the flow that lets the user to finalise the message before sharing it on Facebook.

### Actual results

Facebook iOS app (437.2) does not crash when opened to share content to the feed.

### Steps to reproduce

Facebook iOS app (437.2)
iPhone 12, iOS 17; iPhone 13, iOS 16.5.1.
Facebook IOS SDK version 16.2.1 and 16.2.0.

0. Install Facebook iOS app on your device
1. Download Facebook IOS SDK samples for [16.2.1 sdk version](https://github.com/facebook/facebook-ios-sdk/releases/tag/v16.2.1)
2. Go to samples > FacebookShareSample
3. Add missing FacebookAppID, FacebookClientToken and URL Scheme
4. Change ShareDialog mode [here](https://github.com/facebook/facebook-ios-sdk/blob/main/samples/FacebookShareSample/FacebookShareSample/ShareViewController.swift#L67-L74) to [.native](https://github.com/facebook/facebook-ios-sdk/blob/main/FBSDKShareKit/FBSDKShareKit/UserInterface/ShareDialogMode.swift#L24)
5. Run FacebookShareSample app on your device and try to share link.

**Actual result:** Facebook iOS app is crashing when launched.
**Expected result:** Facebook iOS app opens the flow that lets the user to finalise the message before sharing it on Facebook.

Note: Facebook iOS SDK version 12.1.0 does not crash Facebook iOS when going through the same sharing flow.

### Code samples & details

```swift
// replace https://github.com/facebook/facebook-ios-sdk/blob/main/samples/FacebookShareSample/FacebookShareSample/ShareViewController.swift#L67-L74
    func dialog(withContent content: SharingContent) -> ShareDialog {
        let dialog = ShareDialog(
            viewController: self,
            content: content,
            delegate: self
        )
        dialog.mode = .native
        return dialog
    }
```

**Comment by heweidong2019** [comment]
I had the same problem, hot to fix it?

**Comment by TatyanaLeschenok** [comment]
It looks like the issue has been fixed with Facebook app version 458.0.2 .
